framework/math/PETScUtils/petsc_utils_04_vecmanip.cc

```diff
@@ -2,54 +2,72 @@
 
 #include "chi_log.h"
 
-//###################################################################
+// ###################################################################
 /**Copies a PETSc vector to a STL vector. Only the local portion is
  * copied.*/
-void chi_math::PETScUtils::CopyVecToSTLvector(
-  Vec x, std::vector<double>& data, size_t N)
+void chi_math::PETScUtils::CopyVecToSTLvector(Vec x,
+                                              std::vector<double>& data,
+                                              size_t N,
+                                              bool resize_STL /*=true*/)
 {
-  data.clear();
-  data.resize(N,0.0);
+  if (resize_STL)
+  {
+    data.clear();
+    data.assign(N, 0.0);
+  }
+
   const double* x_ref;
-  VecGetArrayRead(x,&x_ref);
+  VecGetArrayRead(x, &x_ref);
 
-  for (size_t i=0; i<N; ++i)
+  for (size_t i = 0; i < N; ++i)
```

Comment:
std::copy might be better here.

Reply:
Could you give some protocode here? No idea how to do this with that pointer-based vector

Reply:
How does one do that with the double* business?

Reply:
Maybe try `std::copy(x_ref, x_ref + N, data.begin());`?

Reply:
Done